packages/dx-react-scheduler-material-ui/src/templates/appointment-form/recurrence/radio-group/yealy-editor.jsx

```diff
@@ -0,0 +1,271 @@
+
+const styles = ({ spacing }) => ({
+  textEditor: {
+    width: 'calc((100% - 5.5em) * 4 / 7)',
+    minWidth: 'calc(100% - 13.5em)',
+    marginLeft: '1em',
+  },
+  label: {
+    width: '4.5em',
+  },
+  input: {
+    paddingBottom: spacing(2.75),
+  },
+  select: {
+    width: 'calc((100% - 5.5em) * 3 / 7)',
+    maxWidth: '8em',
+  },
+  longSelect: {
+    width: 'calc((100% - 5.5em) * 4 / 7)',
+    minWidth: 'calc(100% - 13.5em)',
+    marginLeft: '1em',
+  },
+  formControlLabel: {
+    alignItems: 'flex-start',
+  },
+  formControl: {
+    marginRight: 0,
+    marginTop: spacing(1),
+    marginBottom: spacing(1),
+  },
+  doubleSelect: {
+    marginLeft: '4.5em',
+    width: 'calc(100% - 4.5em)',
+    marginTop: spacing(1),
+  },
+  radioButton: {
+    marginTop: spacing(0.75),
+  },
+  controlLabel: {
+    width: '100%',
+  },
+});
+
+const getCurrentMonth = (recurrenceOptions, appointmentData) => {
+  if (recurrenceOptions.bymonth) {
+    return recurrenceOptions.bymonth;
+  }
+  return appointmentData.startDate.getMonth() + 1;
+};
+
+const YearlyEditorBase = ({
+  classes,
+  getMessage,
+  labelComponent: Label,
+  textEditorComponent: TextEditor,
+  selectComponent: Select,
+  readOnly,
+  appointmentData,
+  formatDate,
+  onFieldChange,
+  ...restProps
+}) => {
+  const [dayNumber, setDayNumber] = useState(appointmentData.startDate.getDate());
+  const [stateWeekNumber, setStateWeekNumber] = useState(
+    Math.trunc((appointmentData.startDate.getDate() - 1) / 7),
+  );
+  const [stateDayOfWeek, setStateDayOfWeek] = useState(appointmentData.startDate.getDay());
+
+  const { rRule } = appointmentData;
+  const recurrenceOptions = React.useMemo(() => getRecurrenceOptions(rRule) || {}, [rRule]);
+  const changeByMonthDay = React.useCallback(nextByMonthDay => onFieldChange({
+    rRule: handleStartDateChange(nextByMonthDay, recurrenceOptions),
+  }), [recurrenceOptions]);
+
+  const {
+    dayOfWeek, weekNumber, dayNumberTextField, radioGroupValue: value,
+  } = getRadioGroupDisplayData(
+    recurrenceOptions, stateDayOfWeek, stateWeekNumber, dayNumber, 'onDayAndMonth', 'onDayOfWeek',
+  );
+  const month = getCurrentMonth(recurrenceOptions, appointmentData);
+
+  const changeMonth = React.useCallback(nextMonth => onFieldChange({
+    rRule: changeRecurrenceOptions({
+      ...recurrenceOptions, bymonth: nextMonth,
+    }),
+  }), [recurrenceOptions]);
+  const months = React.useMemo(() => getMonths(formatDate), [formatDate]);
+  const monthsWithOf = React.useMemo(
+    () => getMonthsWithOf(getMessage, formatDate), [getMessage, formatDate],
+  );
+
+  const changeWeekNumber = React.useCallback(nextWeekNumber => onFieldChange({
+    rRule: handleWeekNumberChange(nextWeekNumber, recurrenceOptions),
+  }), [recurrenceOptions]);
+  const weekNumbers = React.useMemo(
+    () => getWeekNumberLabels(getMessage), [getMessage],
+  );
+
+  const changeDayOfWeek = React.useCallback(nextDayOfWeek => onFieldChange({
```

Comment:
We should consider extracting editors to separate components - in this component we should only assemble them